### ISSUE TITLE
Unable to find file: run_loop_fast_uia.js

### ISSUE THREAD

**Issue by jmoody** [calabash/run_loop#361]
Happens randomly.

```
Unable to find file: run_loop_fast_uia.js
./features/support/01_launch.rb:42:in `Before'
LaunchControl.launcher.relaunch(launch_options)
LaunchControl.launcher.calabash_notify(self)
```

**Comment by jmoody** [comment]
Also happening in CI:

```
18:32:47 Failures:
18:32:47 
18:32:47   1) Launcher:  #console_attach #attach can connect to launched apps preferences
18:32:47      Failure/Error: return RunLoop.run(args)
18:32:47 
18:32:47      RunLoop::Xcrun::TimeoutError:
18:32:47        Xcrun timed out after 30.23 seconds executing
18:32:47 
18:32:47        xcrun instruments -s templates
18:32:47 
18:32:47        with a timeout of 30
18:32:47      # ./lib/calabash-cucumber/launcher.rb:691:in `block in new_run_loop'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:689:in `times'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:689:in `new_run_loop'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:557:in `relaunch'
18:32:47      # ./spec/integration/launcher/console_attach_spec.rb:84:in `block (5 levels) in <top (required)>'
18:32:47 
18:32:47   2) Launcher:  #console_attach #attach can connect to launched apps host
18:32:47      Failure/Error: return RunLoop.run(args)
18:32:47 
18:32:47      RuntimeError:
18:32:47        Unable to find file: run_loop_host.js
18:32:47      # ./lib/calabash-cucumber/launcher.rb:691:in `block in new_run_loop'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:689:in `times'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:689:in `new_run_loop'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:557:in `relaunch'
18:32:47      # ./spec/integration/launcher/console_attach_spec.rb:84:in `block (5 levels) in <top (required)>'
18:32:47 
18:32:47   3) Launcher:  #console_attach #attach can connect to launched apps shared_element
18:32:47      Failure/Error: return RunLoop.run(args)
18:32:47 
18:32:47      RuntimeError:
18:32:47        Unable to find file: run_loop_shared_element.js
18:32:47      # ./lib/calabash-cucumber/launcher.rb:691:in `block in new_run_loop'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:689:in `times'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:689:in `new_run_loop'
18:32:47      # ./lib/calabash-cucumber/launcher.rb:557:in `relaunch'
18:32:47      # ./spec/integration/launcher/console_attach_spec.rb:84:in `block (5 levels) in <top (required)>'
```

**Comment by NickLaMuro** [comment]
Hi @jmoody,

I was having the same issue as you, and was able to downgrade to [v2.0.1](https://github.com/calabash/run_loop/compare/v2.0.1...v2.0.2) and the error went away entirely.  I was getting it pretty regularly though after the first scenario ran, so it seemed consistent from my end.

Based on the changes, it seems like #261 might be the culprit, but since I am unfamiliar with the gem, I would have to spend some time digging into the changes to find the exact issue.

For now, for those that run into this issue, it seems like you are able to explicitly include the `v2.0.1` version of this gem in the Gemfile and the problem goes away.  I will see if I can find a fix later when I have some free time.

Thanks!

**Comment by jmoody** [comment]
@NickLaMuro Are you passing any options to `relaunch`?

**Comment by NickLaMuro** [comment]
@jmoody Nope, it is an empty hash (just what was generated by `calabash-ios gen`):

```
Before do |scenario|
  launcher = Calabash::Launcher.launcher
  options = {
    # Add launch options here.
  }

  launcher.relaunch(options)
  launcher.calabash_notify(self)
end
```

**Comment by jmoody** [comment]
### Notes

* This is insanely hard to reproduce locally.
* I am ssh into our CI server where I can get it to pop fairly regularly - about once an hour.
* The `validate_script` method is only called when the run-loop options contain the `:script` key.
* The problem is that the `validate_script` method is only being passed the file name, not the path.

**Comment by NickLaMuro** [comment]
Correction:  Now that I have started trying to debug this... it is now happening randomly for me as well...

**Comment by NickLaMuro** [comment]
@jmoody So I was able to find a way to reproduce it, and I was able to test it against both 2.0.1 and 2.0.2 and get different results.

When running against a simulator locally (this is on iOS with Xcode 6), I would wait until the app in the simulator just launched, and then kill the simulator with `CMD+Q`.  This has to be before the suite tries to run any of the steps, otherwise it will just error out with "Connection Refused" (this might be able to be killed sooner, but this is what worked for me).  After a timeout, this would then re-trigger the `RunLoop.run` based on the code here:

https://github.com/calabash/calabash-ios/blob/develop/calabash-cucumber/lib/calabash-cucumber/launcher.rb#L690-L700

I put a print statement in my local copy of the gem for the `args` variable, and what I found is that the `:script` key was not set on the first run, but after a timeout and retry, it was set, probably by some of the following setup scripts that manipulated the hash when there were no defaults set.

With 2.0.1, the `:script` value was a absolute path when it was set on the second run, so the second attempt worked out fine.  On 2.0.2, the value was just the file name, so it failed because it didn't know where the file was on the system.

We could probably fix this in the `calabash-ios` by doing the following in the `calabash-cucumber` gem:

```diff
  def new_run_loop(args)

    last_err = nil

    num_retries = args[:launch_retries] || 5

    num_retries.times do
      begin
-        return RunLoop.run(args)
+        return RunLoop.run(args.clone)
      rescue RunLoop::TimeoutError => e
        last_err = e
        if full_console_logging?
          puts 'retrying run loop...'
        end
        self.simulator_launcher.stop
      end
    end
    self.simulator_launcher.stop
    puts "Unable to start. Make sure you've set APP_BUNDLE_PATH to a build supported by this simulator version"
    raise StartError.new(last_err)
  end
```

But I am not familiar enough with the project to know if that will have any negative side affects.  The other option would be to determine where that variable is being set in the hash when it is not present, and make sure it always sets it as an absolute path, but I have found where that is being done yet... :disappointed:

**Comment by jmoody** [comment]
Yes.  I discovered this today as well.

I will have a solution and a release tomorrow.

Good sleuthing.

**Comment by NickLaMuro** [comment]
Cool, thanks for your work on this!

**Comment by jmoody** [comment]
Hotfix is in progress.

**Comment by kaorijp** [comment]
@jmoody 
I also have same problem 
```
Unable to find file: run_loop_fast_uia.js
```
When will you release the hot fix for this?

**Comment by jmoody** [comment]
Today.  Until then, use run-loop 2.0.1

**Comment by jmoody** [comment]
Released 2.0.3
